lookout/style/typos/preparation.py

```diff
@@ -31,6 +33,108 @@ def _download_url(url: str, output_path: str) -> None:
+    :param frequencies_path: Path to the .csv file with space-separated word-frequency pairs one-per-line.
+    :param config: Configuration for the vocabulary creation:
+                   stable: How much tokens, which don't have more frequent edit-distance-neighbors, to take into \
+                           the vocabulary.
+                   suspicious: How much tokens, whose more frequent edit-distance-neighbor is and English word, \
+                               to take into the vocabulary.
+                   non_suspicious: How much tokens, whose more frequent edit-distance-neighbor is and English word, \
+                                   to take into the vocabulary.
+    :return: Dictionary with the vocabulary tokens as keys and their corresponding frequencies as values.
+    """
+    checker = SymSpell(max_dictionary_edit_distance=2, prefix_length=100)
+    checker.load_dictionary(frequencies_path)
+    frequencies = read_frequencies(frequencies_path)
+    sorted_frequencies = list(sorted(frequencies.items(), key=lambda x: -x[1]))
+
+    # For every token, find a token on edit distance 1, which has higher frequency, if there is one
+    def _correct_token(token_freq):
+        token, freq = token_freq
+        suggestions = checker.lookup(token, 2, 1)
+        if len(suggestions) > 1:
+            correction = suggestions[1].term
+            return correction, frequencies[correction]
+        else:
+            return token, freq
+    corrections = list(tqdm(map(_correct_token, sorted_frequencies), total=len(sorted_frequencies)))
+
+    all_tokens = pandas.DataFrame(columns=["token", "token_freq", "correction", "correction_freq"])
+    all_tokens["token"] = [token for token, _ in sorted_frequencies]
+    all_tokens["token_freq"] = [freq for _, freq in sorted_frequencies]
+    all_tokens["correction"] = [token_freq[0] if token_freq[1] > sorted_frequencies[i][1] else sorted_frequencies[i][0]
+                                for i, token_freq in enumerate(corrections)]
+    all_tokens["correction_freq"] = [token_freq[1] if token_freq[1] > sorted_frequencies[i][1] else sorted_frequencies[i][1]
+                                     for i, token_freq in enumerate(corrections)]
+    all_tokens["rel"] = all_tokens["correction_freq"] / all_tokens["token_freq"]
+
+    # Find all English words among all the tokens
+    eng_voc = set()
+    with tempfile.NamedTemporaryFile() as temp_file:
+        _download_url("https://raw.githubusercontent.com/dwyl/english-words/master/words_alpha.txt", temp_file.name)
+        with open(temp_file.name, "r") as f:
+            for line in f:
+                eng_voc.add(line.strip())
+
+    # Leave only non-english tokens for analysis
+    stable = all_tokens[(all_tokens.rel == 1.0) & ~all_tokens.token.isin(eng_voc)]
+    unstable = all_tokens[(all_tokens.rel > 1) & ~all_tokens.token.isin(eng_voc)]
+
+    # Get tokens and their corrections lemmas
+    os.system("python3 -m spacy download en")
```

Comment:
Wow this is a big hack :) First of all, you should always use subprocess.
Second, sys.executable instead of hardcoding python3. Third, you can do the same by a Python import and running the function inplace.